### PR TITLE
fix: added validation to putAll method

### PR DESCRIPTION
**Pull request by murali-shris**
**- What I did**
- added key validation to hive keystore putAll method
**- How I did it**
- modified implementation of hive_keystore.dart --> putAll method
**- How to verify it**
- run the unit test hive_keystore_impl_test.dart

### CODE REVIEW

**Comment by murali-shris** [comment]
at client tests are successful with key validation branch of persistence
https://github.com/atsign-foundation/at_client_sdk/pull/670/

File: at_secondary/at_persistence_secondary_server/test/hive_keystore_impl_test.dart [hunk 1]
```diff
@@ -208,6 +208,25 @@ void main() async {
           throwsA(predicate((dynamic e) => e is InvalidAtKeyException)));
     });
 
+    test('test create non reserved key- keystore putAll', () async {
+      var keyStoreManager = SecondaryPersistenceStoreFactory.getInstance()
+          .getSecondaryPersistenceStore('@test_user_1')!;
+      var keyStore = keyStoreManager.getSecondaryKeyStore()!;
+      var atData = AtData();
+      atData.data = '123';
+      expect((() async => await keyStore.put('privatekey:mykey', atData)),
```
Comment (gkc) [inline at_secondary/at_persistence_secondary_server/test/hive_keystore_impl_test.dart]: Should we use expectLater instead?

File: at_secondary/at_persistence_secondary_server/test/hive_keystore_impl_test.dart [hunk 2]
```diff
@@ -208,6 +208,25 @@ void main() async {
...
+          throwsA(predicate((dynamic e) => e is InvalidAtKeyException)));
+    });
+
+    test('test putAll invalid key', () async {
+      var keyStoreManager = SecondaryPersistenceStoreFactory.getInstance()
+          .getSecondaryPersistenceStore('@test_user_1')!;
+      var keyStore = keyStoreManager.getSecondaryKeyStore()!;
+      var atData = AtData();
+      atData.data = '123';
+      expect((() async => await keyStore.put('hello@', atData)),
```
Comment (gkc) [inline at_secondary/at_persistence_secondary_server/test/hive_keystore_impl_test.dart]: Should we use expectLater instead?

File: at_secondary/at_persistence_secondary_server/test/hive_keystore_impl_test.dart (hunk 1 above)
Comment (murali-shris) [reply at_secondary/at_persistence_secondary_server/test/hive_keystore_impl_test.dart]: changed

File: at_secondary/at_persistence_secondary_server/test/hive_keystore_impl_test.dart (hunk 2 above)
Comment (murali-shris) [reply at_secondary/at_persistence_secondary_server/test/hive_keystore_impl_test.dart]: changed

**Review by sitaram-kalluri — APPROVED** [review]

**Pull request merged**